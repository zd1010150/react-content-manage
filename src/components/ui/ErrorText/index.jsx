/**
 * This component is for display error message and use the similar style in antd form validation.
 * The reason we create this common component is because in antd design error text need a higher component with a 'hasError' class to have correct style.
 */
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';
import React from 'react';
import { injectIntl, intlShape } from 'react-intl';
import styles from './index.less';

const cx = classNames.bind(styles);


const defaultProps = {
  intlId: 'global.errors.required',
};
const propTypes = {
  intl: intlShape.isRequired,
  intlId: PropTypes.string,
};


const ErrorText = ({ intl, intlId }) => {
  const { formatMessage } = intl;

  return (
    <div className={cx('errorText')}>
      {formatMessage({ id: intlId })}
    </div>
  );
};


ErrorText.defaultProps = defaultProps;
ErrorText.propTypes = propTypes;
export default injectIntl(ErrorText);
