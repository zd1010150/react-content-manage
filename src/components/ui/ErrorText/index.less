@import "~assets/less/theme.less";

.errorText {
  color: @error-color;
  line-height: 21px;
}