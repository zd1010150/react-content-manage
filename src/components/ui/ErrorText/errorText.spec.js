import React from 'react';
import ErrorText from './index';
import renderer from 'react-test-renderer';
import { IntlWrapper } from 'components/tests/index';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <IntlWrapper locale={'zh'}>
        <ErrorText />
      </IntlWrapper>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
