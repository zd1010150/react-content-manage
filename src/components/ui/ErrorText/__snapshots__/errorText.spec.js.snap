// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="errorText"
>
  请输入值！
</div>
`;
